src/pages/blog/dynamic-components-in-your-mpa.mdx

```diff
@@ -0,0 +1,248 @@
+
+## Scenario 3: Your stack can't meet your needs.
+
+After reviewing the existing tools available in your stack and considering tools like HTMX, you may still find your options lacking for what you are being tasked to build. So instead, you want to bring in an SPA-like JavaScript component development flow alongside your existing application. So does this mean it's time to bring in React? There's still some slightly more ideal options to consider first. The challenge with React is that there's no natural interface between server-generated HTML and rendering a declared React component. But is there a platform which offers a natural interface between HTML & rendering a declared component? Thankfully yes! There are several frameworks which run on the back of the [web-component](https://developer.mozilla.org/en-US/docs/Web/API/Web_components) architecture which provide us with exactly such a model.
+
+### Lit
+
+Lit is a framework by Google which allows you to make web-components in a simple and standardized fashion. The key advantage of having your components registered as web-components is that the process of mounting your components happens naturally in the browser's custom elements API. However, the actual development experience is very akin to developing with React class components, rendering using template strings rather than JSX.
+
+<Card>
+```ts
+import { LitElement, html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+
+@customElement("lit-counter")
+export class MyElement extends LitElement {
+  @property({ type: Number })
+  count = 0;
+
+  private _increment() {
+    this.count++;
+  }
+
+  render() {
+    return html`
+      <button @click=${this._increment}>The count is ${this.count}</button>
+    `;
+  }
+}
+```
+<strong style={{ textAlign: "center", display: "block", paddingLeft: "1em", paddingRight: "1em" }}>
+  _Figure 3._ A counter component in Lit
+</strong>
+</Card>
+
+As long as the bundled JavaScript output is added to any & all pages which need these components within your MPA, then all you have to do to add this component is put `<lit-counter></lit-counter>` in your HTML and the component will be naturally instantiated by the browser. So that means no added script tags which target a `<div>` with a particular `#id` on it. Just use your components in a natural fashion.
+
+### Stencil
+
+Stencil is a very similar tool to Lit with slightly different design decisions & UX. Stencil aims to be slightly closer to React, and thus uses JSX. However, it also uses the same concept of decorators which denote reactive properties which trigger a rerender.
+
+<Card>
+```tsx
+import { Component, h, State } from '@stencil/core';
```

Comment:
Are `h` and `State` needed in this example? Btw I love the consistent simple counter example for each option.

Reply:
State actually should be `Prop`, I forgot I changed that. As for `h`, that's the JSX function so it technically is needed in Stencil so I think it's worth leaving in